docs/curl/datatx.md

```diff
@@ -1,213 +1,412 @@
+MEUCIQDm6dLmAdd0X49JygTiUkh+brZxprWSr2+hcAH+QIu3AAIgF+m7kO33YXyyqSbnXS9HR79wt/aL3JGhKvXFQaFBJms=
+```
+Second, submit the query
 ```sh
 curl \
      -H "Content-Type: application/json" \
-     -H "UserID: user1" \
-     -H "Signature: abcd" \
-     -X GET http://127.0.0.1:6001/data/bdb/key1 | jq .
+     -H "UserID: bob" \
+     -H "Signature: MEUCIQDm6dLmAdd0X49JygTiUkh+brZxprWSr2+hcAH+QIu3AAIgF+m7kO33YXyyqSbnXS9HR79wt/aL3JGhKvXFQaFBJms=" \
+     -X GET http://127.0.0.1:6001/data/db2/key1 | jq .
 ```
+
 **Output**
 ```json
 {
-  "payload": {
+  "response": {
     "header": {
-      "nodeID": "bdb-node-1"
+      "node_id": "bdb-node-1"
     },
-    "value": "dmFsdWUy",
+    "value": "eXl5",
     "metadata": {
       "version": {
-        "block_num": 28
+        "block_num": 5
       },
       "access_control": {
         "read_users": {
-          "user2": true
+          "alice": true,
+          "bob": true
         },
         "read_write_users": {
-          "user1": true
+          "alice": true
         }
       }
     }
-  }
+  },
+  "signature": "MEUCID1+8HiN9nkv8990SytuQRl8BhBV2xUEe5InsPB5D7IwAiEAx8qENDc9BQTO3arlOxPWf9lh7OP8xXFoDS+jipnAA2Y="
 }
 ```
+The result contains the `value` associated with the key and also the `access_control` and `version` as part of the
+`metadata`.
 
 ## Updating an existing state
 
 Let's update the value of `key1`. In order to do that, we need to execute the following three steps:
 
 1. Read the `key1` from the cluster.
 2. Construct the updated value and transaction payload including the read version.
-3. Submit the data transaction 
+3. Submit the data transaction
 
-> Note that the `data_reads` should contain the version of the `key1` that was read before modifying the value of `key1`.
+> Note that the `data_reads` should contain the version of the `key1` that was read before modifying the value of `key1`. If `data_reads` is kept empty, the `data_writes` would be considered as blind write.
 
 ```json
  curl \
    -H "Content-Type: application/json" \
-   -X POST http://127.0.0.1:6001/data \
+   -H "TxTimeout: 2s" \
+   -X POST http://127.0.0.1:6001/data/tx \
    --data '{
 	"payload": {
-		"userID": "user1",
-		"txID": "1b6d6414-9b58-45d0-9723-1f31712add81",
-		"DB_name": "bdb",
-        "data_reads": [
-			{
-				"key": "key1",
-				"version": {
-					"block_num": 29,
-					"tx_num": 0
-				}
-			}
+		"must_sign_user_ids": [
+            "alice"
         ],
-		"data_writes": [
-			{
-				"key": "key1",
-				"value": "aXDUvZio",
-                "ACL": {
-					"read_users": {
-						"user2": true
-					},
-					"read_write_users": {
-						"user1": true
-					}
-				}
-			}
+		"tx_id": "1b6d6414-9b58-45d0-9723-1f31712add83",
+        "db_operations": [
+            {
+                "db_name": "db2",
+                "data_reads": [
+                    {
+                        "key": "key1",
+                        "version": {
+                            "block_num": 5
+                        }
```

Comment:
block+index

Reply:
Due to `omitempty`. https://github.com/IBM-Blockchain/bcdb-server/issues/109